### PR TITLE
updating Python syntax to latest group standard

### PR DESCRIPTION
**Pull request by gforsyth**
This is the result of running a script to automatically update the formatting across every notebook in the group.  

The following has been done automatically:

`import matplotlib.pyplot as plt` -> `from matplotlib import pyplot`
`import numpy as np` -> `import numpy`

Changing all instances of np. and plt. to numpy. and pyplot.

The following needs manual intervention:

*  [ ] Tweaking the writing/descriptions that mention these things
*  [ ] Changing all `from math import *` to instead be `from math import [needed math functions]`

### CODE REVIEW

**Comment by ahmadia** [comment]
Huh?  `import numpy as np` and `matplotlib.pyplot as plt` are part of the standard scientific computing lexicon these days.  I guess the decision has already been made, and I think I understand the reasons for doing it, but I don't think this is a good idea.

File: lessons/00_Lesson00_QuickPythonIntro.ipynb [hunk 1]
```diff
@@ -82,8 +82,8 @@
      "input": [
       "# <-- comments in python are denoted by the pound sign, like this one\n",
       "\n",
-      "import numpy as np                 # we import the array library, and call it 'np'\n",
-      "import matplotlib.pyplot as plt    # import plotting library, and call it 'plt'"
+      "import numpy                 # we import the array library, and call it 'np'\n",
```
Comment (ahmadia) [inline lessons/00_Lesson00_QuickPythonIntro.ipynb]: comments need to be updated here

**Comment by gforsyth** [comment]
@ahmadia -- agreed that they are standard, but they're not really a very good standard.  

I pushed for this because I think it quickly breaks down -- e.g. 

`import scipy as sp`
`import sympy as ?`

I'm interested in your thoughts on it.  Is there something I'm overlooking besides the obvious taboo of going against the predominant convention?

**Comment by ahmadia** [comment]
You have to recognize that NumPy is used *everywhere*.  In the past, there was a bad convention of doing `from numpy import *` or the `from pylab import *` shortcut that really polluted the namespace.  

As far as I know, sympy tends to want to be called sympy or sym, and scipy's standard is sp.  Another example is pandas -> pd.  If somebody searches on Stack Overflow, they are [5x](https://encrypted.google.com/search?hl=en&q=np.linspace%20site%3Astackoverflow.com#hl=en&q=%22np.linspace%22+site:stackoverflow.com) as [likely](https://encrypted.google.com/search?hl=en&q=np.linspace%20site%3Astackoverflow.com#hl=en&q=%22numpy.linspace%22+site:stackoverflow.com) to turn up `np.linspace` than `numpy.linspace`.  

I could come up with other examples, but going against the predominant convention when teaching beginners to program is generally a bad idea unless you're going to be willing to spend the extra time explaining to them what you're doing differently and why.

**Comment by ahmadia** [comment]
I should clarify that `numpy` and `pyplot` are the only two (of the libraries that your code uses) that I have really strong feelings about.  There aren't strong enough conventions for the others that you're going to risk causing confusion.  I agree that `sympy`'s import is a bit muddled.

**Comment by gforsyth** [comment]
We certainly are teaching beginners, especially with NumericalMOOC, but I would posit that beginners should really understand what it means to be using a function from a library before they continue.  That might be a little bit pie-in-the-sky, but the only thing I see this as potentially mucking up is people aggregating bits of code from here and there without understanding what they're doing.  

I would also argue that the readability of code using the full module names is improved which can be a great help to beginners.  

We are also following the [Google Style Guide](https://google-styleguide.googlecode.com/svn/trunk/pyguide.html?showone=Imports#Imports) (for whatever that's worth... probably nothing)

**Comment by ahmadia** [comment]
It comes down a little to whether you believe that equations and code should use shorthand: `v` for velocity, `x` for the position of the coordinate along the first axis of the Cartesian plane, etc...  There are times when it's important to be verbose, and there are times where it's equally important to be concise.  I'm not going to go any deeper into this with you since it doesn't look like we're changing each other's minds, but you might consider bringing this topic up for discussion on the Software Carpentry forums.

FWIW, @gvwilson appears to agree with you, at least 50% of the time :)  His first two lessons explicitly `import numpy`, but the third lesson switches to the shorthand with little explanation...

http://software-carpentry.org/v5/novice/python/03-loop.html

```
%matplotlib inline

import numpy as np
from matplotlib import pyplot as plt
```

**Comment by gforsyth** [comment]
I'm stubborn by nature.  :)

Verbosity and concision are great and the squeeze theorem (misapplied) tells us that there has to be a line somewhere that divides them, but I don't think that the extra 3 characters push this style of import into the realm of verbosity.  

I'll bring it up in the SC forums, as you suggest.

**Comment by gvwilson** [comment]
A PR inserting explanation would be welcome :-)

**Comment by ahmadia** [comment]
@gforsyth - A PR or issue against Greg's lesson (explaining why the import statement has changed to the shorthand form) might be the right place to start the conversation.  There's a general "Discussion" mailing list that we try to use to point people at discussions happening on GitHub.

**Comment by gforsyth** [comment]
Ok, I'll do that, then.  We're about to have a weekly meeting, so I'm going to hold off for today -- I imagine I'll want to be around to field the various responses / projectile fruit.

**Comment by labarba** [comment]
> the only thing I see this as potentially mucking up is people aggregating bits of code from here and there without understanding what they're doing.

This comment by @gforsyth makes me even more enthusiastic for our against-the-current convention! If "our way" makes students more likely to read through the code they're picking up from StackOverflow to fix the function calls to NumPy, then all the better.

Internally, we've argued this ad nauseam and finally settled on the full `numpy` prefix being more readable and the `import numpy` command more natural and easy to explain to beginners.

Enjoy this Twitter exchange!
https://twitter.com/LorenaABarba/status/545342446407020545

@mesnardo --- could you review Gil's changes and merge the PR if it passes?
(I'll assign you now!)

**Comment by mesnardo** [comment]
In the first lesson [QuickPythonIntro](http://nbviewer.ipython.org/github/barbagroup/AeroPython/blob/master/lessons/00_Lesson00_QuickPythonIntro.ipynb), we should explain a little bit more why we deviate from the common convention, and why we believe our way is the correct one, adding a reference to the [Google Style Guide](https://google-styleguide.googlecode.com/svn/trunk/pyguide.html) webpage.

Also, changing
```python
from math import *
```
to
```python
from math import sin, cos
```
is not consistent with the way we import other modules.

To be fully consistent, we should have:
```python
import math
```
and use the mathematical function with `math.cos()`. Remember that Numpy has also a function `cos`.

**Comment by ahmadia** [comment]
This all sounds good.  I don't think the link to the Google Style Guide is particularly necessary, it is not what I would consider highly regarded in the scientific Python developer community.

**Comment by gforsyth** [comment]
@ahmadia Similar thoughts on the PEP8 style guide?  Or just leave out style guide references completely?

**Comment by ahmadia** [comment]
I think you should link to PEP8, which is definitely followed where possible.  You can quote "explicit is better than implicit" from there as justification :)

**Comment by gforsyth** [comment]
@mesnardo 

I don't think that 

    from math import sin, cos

is inconsistent.  We do

    from matplotlib import pyplot

I think it's fine to do `import math` and then use `math.cos()` but I also think that if we only need one or two functions from a library then it is ok to just grab those that we need, e.g. `from matplotlib import cm` for colormaps.  

Having said all of that, it's good to help new users distinguish between `math.sin()` and `numpy.sin()` as you suggest.  The overhead of `numpy.sin()` is minimal but if we know that we should be dealing with a scalar then `math.sin()` serves as a nice check.

**Comment by gforsyth** [comment]
That is a good point.  Also, @mesnardo points out (from across the room) that `cm` is a module anyway.

The issue we're debating now is that -- in an effort to be consistent but also not be a huge pain in the butt -- this material in particular uses a bunch of scalar trig and will it negatively impact readability...  blarg

**Comment by ahmadia** [comment]
This is why I recommended `import numpy as np` in the first place :)  However, I think it's also okay to `from numpy import x, y, z` as long as you are consistent in explaining what you're doing.

**Comment by ahmadia** [comment]
By the way, this isn't worth obsessing over.  Just pick something and go with it :)

**Comment by labarba** [comment]
I think we've debated and agreed, finally, so this can be merged now?

File: lessons/00_Lesson00_QuickPythonIntro.ipynb [hunk 2]
```diff
@@ -94,7 +94,7 @@
      "cell_type": "markdown",
      "metadata": {},
      "source": [
-      "What's all of this `import-as` business? We are importing one library named `numpy` and we are importing a module called `pyplot` of a big library called `matplotlib`.  The two lines above have created shortcuts to those libraries as `np` and `plt`, respectively. The reason for creating the shortcuts is that we need to type the library names quite often in the code.\n",
+      "What's all of this `import-as` business? We are importing one library named `numpy` and we are importing a module called `pyplot` of a big library called `matplotlib`.  \n",
```
Comment (ahmadia) [inline lessons/00_Lesson00_QuickPythonIntro.ipynb]: There's no `import as`, this sentence still needs a little love.

File: lessons/00_Lesson00_QuickPythonIntro.ipynb [hunk 3]
```diff
@@ -172,7 +172,7 @@
      "metadata": {},
      "source": [
       "\n",
-      "Sometimes, you'll see people importing a whole library without assigning a shortcut for it (like `np` here for `numpy`). This saves typing but is sloppy and can get you in trouble. Best to get into good habits from the beginning!\n",
+      "Sometimes, you'll see people importing a whole library without assigning a shortcut for it (like `from numpy import *`). This saves typing but is sloppy and can get you in trouble. Best to get into good habits from the beginning!\n",
```
Comment (ahmadia) [inline lessons/00_Lesson00_QuickPythonIntro.ipynb]: Also needs fixing.

**Comment by ahmadia** [comment]
I haven't gone over this with a fine-toothed comb but it looks good aside from the two sentences I flagged in the comments.

**Pull request merged**